tests/e2e/sms/batches/test_batch_sms.py

```diff
@@ -54,6 +53,21 @@ def test_send_sms_sync(sinch_client_sync, phone_number, origin_phone_number):
         to=[phone_number],
         from_=origin_phone_number,
         body="Synchronous Spanish Inquisition",
-        feedback_enabled=True,
+        feedback_enabled=True
+    )
+    assert isinstance(send_sms_response, SendSMSBatchResponse)
+
+
+def test_send_sms_sync_with_service_plan_id(
+    sinch_client_sync_with_service_plan_id,
+    phone_number,
+    origin_phone_number
+):
+    send_sms_response = sinch_client_sync_with_service_plan_id.sms.batches.send(
+        delivery_report="none",
+        to=[phone_number],
+        from_=origin_phone_number,
+        body="Synchronous Spanish Inquisition",
+        feedback_enabled=True
     )
     assert isinstance(send_sms_response, SendSMSBatchResponse)
```

Comment:
There is no place where we ensure the SendSMSBatchResponse correctly reads the response correctly: it implicitly assumes all the properties are present. I remember requesting numerous times to enhance the tests and we have also decided last September that using wiremock from a badly-named branch on a non-reviewed repo needed to be changed towards the official team's mockserver. I know this PR has been opened a long time before that but there had been multiple changes to it and to quote what I can read in this file: `Now is better than never.` Can you open a PR targeting this branch with e2e tests using the mock server?

Reply:
Wiremock is validating all the payloads URLs etc. If there's some issues test is being interrupted.
I don't have time for mock server integration; I have to leave it to the next dev.
Btw. This is out of the topic? This PR is about service_plan_id integration.

Reply:
I'm not sure to follow: against what wiremock is checking a response payload and do you ensure a property defined in a SDK dataclass is properly deserialized? 

> Btw. This is out of the topic? This PR is about service_plan_id integration.

You are writing new e2e tests in this PR, so instead of investing some time on the deprecated wiremock, it would be preferable to invest it on the new mockserver

Reply:
Let's discuss this on our Extended Discussion meeting today

Reply:
https://tickets.sinch.com/browse/DEVEXP-696